### PR TITLE
feat: integrate stateful scenes with content servers

### PR DESCRIPTION
**Pull request by nchamo**
We are basically doing two things now:
1. When we load a scene in edit mode, we are checking if the content server has the scene's state definition. 
2. When we publish a scene in edit mode, we generate the scene's code, and we deploy it, together will all the content files and the state definition

### CODE REVIEW

File: kernel/packages/scene-system/stateful-scene/RendererStatefulActor.ts [hunk 1]
```diff
@@ -1,39 +1,50 @@
-import { CLASS_ID } from "decentraland-ecs/src";
-import { IEngineAPI } from "shared/apis/EngineAPI";
-import { AttachEntityComponentPayload, ComponentCreatedPayload, ComponentRemovedPayload, ComponentUpdatedPayload, CreateEntityPayload, EntityAction, RemoveEntityPayload, UpdateEntityComponentPayload } from "shared/types";
-import { Component, ComponentData, ComponentId, EntityId, StatefulActor } from "./types"
-import { EventSubscriber } from "decentraland-rpc";
-import { generatePBObjectJSON } from "scene-system/sdk/Utils";
+import { CLASS_ID } from 'decentraland-ecs/src'
+import { IEngineAPI } from 'shared/apis/EngineAPI'
```
Comment (nchamo) [inline kernel/packages/scene-system/stateful-scene/RendererStatefulActor.ts:2]: Just formatting changes 😄

File: kernel/packages/scene-system/stateful-scene/SceneStateDefinitionSerializer.ts [hunk 2]
```diff
@@ -1,63 +1,28 @@
...
-        ?.map((component: any) => ({
-          componentId: humanReadableTypeToId(component.type),
-          data: component.value,
-        }))
-      sceneState.addEntity(id, components)
+import { SerializedSceneState } from 'shared/apis/SceneStateStorageController/types'
+import { SceneStateDefinition } from './SceneStateDefinition'
+import { Component, EntityId } from './types'
+
+export function serializeSceneState(state: SceneStateDefinition): SerializedSceneState {
```
Comment (nchamo) [inline kernel/packages/scene-system/stateful-scene/SceneStateDefinitionSerializer.ts:5]: I did 2 things here:
1. `SerializedSceneState` is now being used to send data from the worker to kernel. It will later be transformed again in order to store it. The idea is to decouple storage format from transfer format
2. Embracing functions

File: kernel/packages/shared/apis/SceneStateStorageController/SceneStateStorageController.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+  @exposeMethod
+  async storeState(sceneId: string, sceneState: SerializedSceneState): Promise<void> {
+    // Convert to storable format
+    const storableFormat = fromSerializedStateToStorableFormat(sceneState)
+
+    if (DEBUG) {
+      saveToLocalStorage(`scene-state-${sceneId}`, storableFormat)
+
+      // TODO: Let the renderer know that the deployment was successful
```
Comment (nchamo) [inline kernel/packages/shared/apis/SceneStateStorageController/SceneStateStorageController.ts:38]: This will be done in another PR, after a needed discussion

**Review by nchamo — COMMENTED** [review]
Adding some context comments

File: kernel/packages/shared/apis/SceneStateStorageController/SceneStateStorageController.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+import { ParcelIdentity } from '../ParcelIdentity'
+import { Store } from 'redux'
+import { RootState } from 'shared/store/rootTypes'
+import { getUpdateProfileServer } from 'shared/dao/selectors'
+import { createGameFile } from './SceneStateDefinitionCodeGenerator'
+
+declare const window: any
+
+@registerAPI('SceneStateStorageController')
+export class SceneStateStorageController extends ExposableAPI {
```
Comment (pablitar) [inline kernel/packages/shared/apis/SceneStateStorageController/SceneStateStorageController.ts:27]: We may eventually want to move all this logic to another place in order to decouple this for the Scene RPC.

I'm actually just guessing that we are going to need that, but nonetheless I wanted to plant the idea haha.

**Review by pablitar — APPROVED** [review]
LGTM! Great work!
